File: VB/MultiSelectionHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.Utils

Namespace WindowsApplication1
	Public Class MultiSelectionHelper

		Private _GridView As GridView
		Public Sub New(ByVal gridView As GridView)
			_GridView = gridView
			InitProperties()
			SubscribeEvents()
		End Sub
		Private Sub InitProperties()
			_GridView.OptionsBehavior.Editable = False
			_GridView.OptionsSelection.MultiSelect = True
			_GridView.OptionsSelection.EnableAppearanceFocusedCell = False
			_GridView.FocusRectStyle = DrawFocusRectStyle.None
		End Sub
		Private Sub SubscribeEvents()
			AddHandler _GridView.MouseDown, AddressOf _GridView_MouseDown
		End Sub

		Private Sub _GridView_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
			OnMouseDown(e)
		End Sub
		Private Sub OnMouseDown(ByVal e As MouseEventArgs)
			Dim hi As GridHitInfo = _GridView.CalcHitInfo(e.Location)
			If (Not hi.InRow) Then
				Return
			End If
			_GridView.FocusedRowHandle = hi.RowHandle
			_GridView.FocusedColumn = hi.Column
			_GridView.InvertRowSelection(hi.RowHandle)
			DXMouseEventArgs.GetMouseArgs(e).Handled = True

		End Sub

	End Class
End Namespace

File: CS/MultiSelectionHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.Utils;

namespace WindowsApplication1
{
	public class MultiSelectionHelper
	{

		private GridView _GridView;
		public MultiSelectionHelper(GridView gridView)
		{
			_GridView = gridView;
			InitProperties();
			SubscribeEvents();
		}
		private void InitProperties()
		{
			_GridView.OptionsBehavior.Editable = false;
			_GridView.OptionsSelection.MultiSelect = true;
			_GridView.OptionsSelection.EnableAppearanceFocusedCell = false;
			_GridView.FocusRectStyle = DrawFocusRectStyle.None;
		}
		private void SubscribeEvents()
		{
			_GridView.MouseDown += _GridView_MouseDown;
		}

		private void _GridView_MouseDown(object sender, MouseEventArgs e)
		{
			OnMouseDown(e);
		}
		private void OnMouseDown(MouseEventArgs e)
		{
			GridHitInfo hi = _GridView.CalcHitInfo(e.Location);
			if (!hi.InRow)
			{
				return;
			}
			_GridView.FocusedRowHandle = hi.RowHandle;
			_GridView.FocusedColumn = hi.Column;
			_GridView.InvertRowSelection(hi.RowHandle);
			DXMouseEventArgs.GetMouseArgs(e).Handled = true;

		}

	}
}
